hw/hw6/hw6_2_oop2.py

```diff
@@ -0,0 +1,186 @@
+"""Homework 6.2."""
+import datetime
+from typing import Optional
+
+
+class DeadlineError(Exception):
+    """Raise an error if the deadline occure."""
+
+    pass
+
+
+class HomeworkDoesNotExistError(Exception):
+    """Raise an error if the homework to be deleted does not exist."""
+
+    pass
+
+
+class UniversityLifeForm:
+    """Set first name, last name of a person."""
+
+    def __init__(self, first_name: str, last_name: str) -> None:
+        self.first_name = first_name
+        self.last_name = last_name
+
+
+class Student(UniversityLifeForm):
+    """Create student."""
+
+    def __init__(self, *args: str) -> None:
+        super().__init__(*args)
+
+    def do_homework(
+        self, homework: "Homework", solution: str
+    ) -> Optional["HomeworkResult"]:
+        """Сheck if homework is done.
+
+        Args:
+            homework: Instance of class Homework
+            solution: solution of task
+
+        Returns:
+            The return value. If the time for completing the task has not expired -
+                              return instance of class HomeworkResult
+                              else - raise DeadlineError: "You are late"
+        """
+        self.homework = homework
+        self.solution = solution
+        self.end_time = datetime.datetime.now()
+        if self.homework.is_active():
+            return HomeworkResult(self, self.homework, self.solution, self.end_time)
+        raise DeadlineError("You are late")
+
+
+class Teacher(UniversityLifeForm):
+    """Create teacher."""
+
+    homework_done = {}
+
+    def __init__(self, *args: str) -> None:
+        super().__init__(*args)
+
+    @staticmethod
+    def create_homework(text: str, days_amount: int) -> "Homework":
+        """Create instance of class Homework.
+
+        Args:
+            text: task text
+            days_amount: amount of days to complete the task
+
+        Returns:
+            The return value. Instance of class Homework
+        """
+        return Homework(text, days_amount)
+
+    @classmethod
+    def check_homework(cls: "Teacher", homework_result: "HomeworkResult") -> bool:
+        """Check homework for more than 5 letters.
+
+            If the condition is met, enter satisfactory results into the dictionary "homework_done", with the key:
+            "instance of class Homework", the value: "instance of class HomeworkResult"
+
+        Args:
+            homework_result: instance
+
+        Returns:
+            The return value. True if the results of the homework check are satisfactory
+                              (the length of the solution is more than 5 letters)
+        """
+        if len(homework_result.solution) > 5:
+            if homework_result.homework in cls.homework_done:
+                cls.homework_done[homework_result.homework] = cls.homework_done[
+                    homework_result.homework
```

Comment:
Can we make this block of code more flatten? Also, note that here we have two return True statements which can be merged